<a href="https://colab.research.google.com/github/e23281-lgtm/Statistical-Learning-e23281/blob/main/Copy_of_fundamentals_of_probability_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Q3) Transformer Failiures

In a certain electricity distribution district, the number of emergency transformer failures reported in a single day is modeled as a discrete random variable $X$. Assume that $X$ takes values in the set $\{0,1,2,3\}$, with probabilities

$$
P(X=0)=0.50,\qquad P(X=1)=0.30,\qquad P(X=2)=0.15,\qquad P(X=3)=0.05.
$$

1. Construct a probability space $(\Omega,\mathfrak F,P)$ suitable for this model.
2. Define $X$ as a random variable on this space.
3. Determine the probability mass function of $X$.
4. Compute the expectation $E[X]$.
5. Compute the variance $\mathrm{Var}(X)$.
6. Find $P(X\ge 1)$ and $P(X\ge 2)$.
7. Briefly explain the practical meaning of the expectation and variance in this setting.



ANSWERS


---



1. Construction of the Probability Space $(\Omega, \mathcal{F}, P)$:
 Sample Space ($\Omega$): $\{0, 1, 2, 3\}$. Each outcome represents the specific number of emergency transformer failures reported in a single day.
 $\sigma$-algebra ($\mathcal{F}$): The power set $\mathcal{P}(\Omega)$, containing $2^4 = 16$ possible events.
 Probability Measure ($P$): Defined by the given mass assignments:
   $$P(\{0\}) = 0.50, \quad P(\{1\}) = 0.30, \quad P(\{2\}) = 0.15, \quad P(\{3\}) = 0.05$$
   
2. Definition of $X$:

$X$ is the identity random variable $X: \Omega \to \mathbb{R}$ where $X(\omega) = \omega$. It maps each outcome in the sample space to its corresponding numerical value of failures.

3. Probability Mass Function (PMF) of $X$:The PMF $p_X(x) = P(X = x)$ is:$p_X(0) = 0.50$$p_X(1) = 0.30$$p_X(2) = 0.15$$p_X(3) = 0.05$$p_X(x) = 0$ for $x \notin \{0, 1, 2, 3\}$

In [1]:
import numpy as np
import plotly.graph_objects as go

# Data Setup
x_values = np.array([0, 1, 2, 3])
probabilities = np.array([0.50, 0.30, 0.15, 0.05])

# 4. Compute Expectation E[X]
# Formula: Sum(x * P(x))
expectation = np.sum(x_values * probabilities)

# 5. Compute Variance Var(X)
# Formula: E[X^2] - (E[X])^2
expectation_sq = np.sum((x_values**2) * probabilities)
variance = expectation_sq - (expectation**2)

# 6. Compute Specific Probabilities
# P(X >= 1) = P(1) + P(2) + P(3) or 1 - P(0)
prob_ge_1 = np.sum(probabilities[x_values >= 1])

# P(X >= 2) = P(2) + P(3)
prob_ge_2 = np.sum(probabilities[x_values >= 2])

# Visualization (Sir's preferred style for Discrete Variables)
fig = go.Figure(data=[
    go.Bar(x=x_values, y=probabilities,
           text=probabilities, textposition='auto',
           marker_color='royalblue', opacity=0.75)
])

fig.update_layout(
    title="Probability Mass Function of Transformer Failures (X)",
    xaxis=dict(title="Number of Failures (x)", tickmode='linear'),
    yaxis=dict(title="P(X=x)", range=[0, 0.6]),
    template="plotly_white",
    width=700, height=450
)
fig.show()

print(f"--- Numerical Results ---")
print(f"4. Expectation E[X]: {expectation:.2f}")
print(f"5. Variance Var(X): {variance:.4f}")
print(f"6. P(X >= 1): {prob_ge_1:.2f}")
print(f"6. P(X >= 2): {prob_ge_2:.2f}")

--- Numerical Results ---
4. Expectation E[X]: 0.75
5. Variance Var(X): 0.7875
6. P(X >= 1): 0.50
6. P(X >= 2): 0.20


7. Practical Meaning:

 Expectation ($E[X] = 0.75$): This represents the long-term average number of failures the district can expect per day. While the most likely outcome on any single day is 0 failures ($50\%$ probability), over many months, the average will settle at $0.75$ failures/day. This value is critical for resource allocation and staffing.

 Variance ($\text{Var}(X) = 0.7875$):

 This measures the "risk" or spread of the failures. A higher variance would imply that the number of failures is unpredictable and fluctuates wildly from the mean. In this setting, it helps the district understand the potential for "bad days" (multiple failures) versus "good days" (zero failures) and how much backup equipment should be kept in inventory.

# Q4) Probability of Defective Items

In a manufacturing process, the proportion of defective items in a large batch produced on a given day is modeled by a continuous random variable $X$. Since $X$ represents a proportion, it takes values in the interval $[0,1]$. Suppose that $X$ has a Beta distribution with parameters $\alpha=2$ and $\beta=18$, so that its density is

$$
f(x)=
\begin{cases}
\dfrac{1}{B(2,18)}x^{1}(1-x)^{17}, & 0<x<1,\\\
0, & \text{otherwise},
\end{cases}
$$

where $B(2,18)$ is the Beta function.

Answer the following:

1. Construct a probability space $(\Omega,\mathfrak F,P)$ suitable for this model.

2. Define $X$ as a continuous random variable on this space.

3. Write down the probability density function of $X$.

4. Verify that $f$ is a valid probability density function.

5. Compute the expectation $E[X]$.

6. Compute the variance $\mathrm{Var}(X)$.

7. Find the probability that the defective proportion is at most $0.10$, that is, compute

$$
P(X\le 0.10).
$$

You may leave your answer in Beta integral form.

8. Find the probability that the defective proportion lies between $0.05$ and $0.15$, that is, compute

$$
P(0.05\le X\le 0.15).
$$

You may leave your answer in integral form.

9. Briefly explain why a Beta distribution is a reasonable model for this situation.

10. Interpret the expectation and variance in the context of production quality.


ANSWERS


---



1. Probability Space $(\Omega, \mathcal{F}, P)$:

Sample Space ($\Omega$): The interval $[0, 1]$, representing all possible proportions of defective items.

$\sigma$-algebra ($\mathcal{F}$): The Borel $\sigma$-algebra $\mathcal{B}([0, 1])$.

Probability Measure ($P$): Defined by the density function $f(x)$ such that for any event $A \in \mathcal{F}$, $P(A) = \int_A f(x) dx$.

2. Definition of $X$:

$X$ is the identity random variable $X: \Omega \to \mathbb{R}$ where $X(\omega) = \omega$. It is a continuous random variable because its range is an interval and its probability is defined by an integral over a density function.

3. Probability Density Function (PDF):$$f(x) = \begin{cases} \frac{1}{B(2, 18)} x^{2-1} (1-x)^{18-1}, & 0 < x < 1 \\ 0, & \text{otherwise} \end{cases}$$where $B(2, 18) = \int_0^1 x^1 (1-x)^{17} dx$.

4. Verification:$f$ is a valid PDF because:

$f(x) \geq 0$ for all $x \in [0, 1]$.$\int_0^1 f(x) dx = \frac{1}{B(2, 18)} \int_0^1 x^1 (1-x)^{17} dx = \frac{B(2, 18)}{B(2, 18)} = 1$.

5. Expectation $E[X]$:For a Beta distribution, $E[X] = \frac{\alpha}{\alpha + \beta} = \frac{2}{2 + 18} = \frac{2}{20} = 0.10$.

6. Variance $\text{Var}(X)$:$$\text{Var}(X) = \frac{\alpha\beta}{(\alpha+\beta)^2(\alpha+\beta+1)} = \frac{2(18)}{(20)^2(21)} = \frac{36}{400 \times 21} = \frac{36}{8400} \approx 0.00429$$

7. Probability $P(X \leq 0.10)$:$$P(X \leq 0.10) = \frac{1}{B(2, 18)} \int_0^{0.10} x(1-x)^{17} dx$$

8. Probability $P(0.05 \leq X \leq 0.15)$:$$P(0.05 \leq X \leq 0.15) = \frac{1}{B(2, 18)} \int_{0.05}^{0.15} x(1-x)^{17} dx$$

In [2]:
import numpy as np
import plotly.graph_objects as go
from scipy.stats import beta

# Parameters
alpha_val, beta_val = 2, 18

# Calculations
mean = alpha_val / (alpha_val + beta_val)
variance = (alpha_val * beta_val) / ((alpha_val + beta_val)**2 * (alpha_val + beta_val + 1))
prob_7 = beta.cdf(0.10, alpha_val, beta_val)
prob_8 = beta.cdf(0.15, alpha_val, beta_val) - beta.cdf(0.05, alpha_val, beta_val)

# Visualization
x = np.linspace(0, 0.4, 500) # Proportion usually stays low
y = beta.pdf(x, alpha_val, beta_val)

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=x, y=y,
    mode='lines',
    line=dict(color='firebrick', width=3),
    fill='tozeroy',
    name='Beta(2, 18) PDF'
))

fig.update_layout(
    title='Beta Distribution of Defective Proportions',
    xaxis=dict(title='Proportion of Defective Items (x)'),
    yaxis=dict(title='Density f(x)'),
    template='plotly_white'
)

fig.show()

print(f"--- Numerical Verification ---")
print(f"5. Expectation E[X]: {mean:.2f}")
print(f"6. Variance Var(X): {variance:.5f}")
print(f"7. P(X <= 0.10): {prob_7:.4f}")
print(f"8. P(0.05 <= X <= 0.15): {prob_8:.4f}")

--- Numerical Verification ---
5. Expectation E[X]: 0.10
6. Variance Var(X): 0.00429
7. P(X <= 0.10): 0.5797
8. P(0.05 <= X <= 0.15): 0.5562


9. Why the Beta distribution is reasonable:

The Beta distribution is naturally bounded between $[0, 1]$, making it ideal for modeling proportions, probabilities, or percentages. Its parameters $\alpha$ and $\beta$ allow for various shapes; specifically, $\alpha=2, \beta=18$ creates a right-skewed distribution, which realistically models a manufacturing process where most batches have a low defect rate, but high defect rates are still possible.

10. Interpretation of Quality:

Expectation ($0.10$): On average, $10\%$ of the items in any given batch are expected to be defective. This represents the "center" of the production quality.

Variance ($\approx 0.0043$):

This reflects the consistency of the manufacturing process. A low variance indicates that the defect rate is stable from day to day. If the variance were to increase, it would suggest the process is becoming unstable and unpredictable, even if the average remains at $10\%$.

# Q5) Expected Project Completion Time

A construction company is estimating the time required to install a backup generator at a hospital. Let $X$ denote the number of hours needed to complete the installation. Based on expert judgment, the shortest possible time is $4$ hours, the longest possible time is $10$ hours, and the most likely completion time is $6$ hours. Assume that $X$ follows a Triangular distribution with parameters $a=4$, $b=10$, and $c=6$.

Answer the following:

1. Specify the support of $X$.

2. Write down the probability density function $f_X(x)$.

3. Verify that $f_X$ integrates to $1$ over its support.

4. Compute the expectation $E[X]$.

5. Compute the variance $\mathrm{Var}(X)$.

6. Interpret the values of $a$, $b$, and $c$ in the context of this problem.

7. Explain briefly why a Triangular distribution is a reasonable model in this setting.

8. Compute the probability that the installation takes at most $5$ hours, that is, compute

$$
P(X \leq 5).
$$

9. Compute the probability that the installation takes between $5$ and $8$ hours, that is, compute

$$
P(5 \leq X \leq 8).
$$

10. Briefly explain how this model could be used in project planning or sensitivity analysis.


ANSWERS

---



1. Support of $X$:

The support is the interval $[a, b]$, which in this case is $[4, 10]$.

2. Probability Density Function $f_X(x)$:For $a=4, b=10, c=6$:$$f_X(x) = \begin{cases} \frac{2(x-4)}{(10-4)(6-4)} = \frac{x-4}{6}, & 4 \leq x < 6 \\ \frac{2(10-x)}{(10-4)(10-6)} = \frac{10-x}{12}, & 6 \leq x \leq 10 \\ 0, & \text{otherwise} \end{cases}$$

3. Verification:The area of a triangle is $\frac{1}{2} \times \text{base} \times \text{height}$.Base $= b - a = 10 - 4 = 6$.Height $= f_X(c) = \frac{2}{b-a} = \frac{2}{6} = \frac{1}{3}$.Area $= \frac{1}{2} \times 6 \times \frac{1}{3} = 1$. Thus, it is a valid PDF.

4. Expectation $E[X]$:$$E[X] = \frac{a+b+c}{3} = \frac{4+10+6}{3} = \frac{20}{3} \approx 6.67 \text{ hours}$$

5. Variance $\text{Var}(X)$:$$\text{Var}(X) = \frac{a^2+b^2+c^2-ab-ac-bc}{18} = \frac{16+100+36-40-24-60}{18} = \frac{28}{18} \approx 1.556$$

6. Interpretation of Parameters:

     $a = 4$: The optimistic time (shortest possible duration).

     $b = 10$: The pessimistic time (longest possible duration).

     $c = 6$: The most likely time (mode).

7. Why the Triangular Distribution is reasonable:It is used when physical data is scarce but expert estimates are available. Unlike a Normal distribution, it has fixed boundaries ($a$ and $b$), which is realistic for project tasks that cannot take "negative" time or "infinite" time.

In [3]:
import numpy as np
import plotly.graph_objects as go
from scipy.stats import triang

# Parameters
a, b, c = 4, 10, 6
loc = a
scale = b - a
c_scaled = (c - a) / (b - a)

# Calculations
mean = (a + b + c) / 3
variance = (a**2 + b**2 + c**2 - a*b - a*c - b*c) / 18
prob_8 = triang.cdf(5, c_scaled, loc, scale)
prob_9 = triang.cdf(8, c_scaled, loc, scale) - triang.cdf(5, c_scaled, loc, scale)

# Visualization
x = np.linspace(3, 11, 500)
y = triang.pdf(x, c_scaled, loc, scale)

fig = go.Figure()
fig.add_trace(go.Scatter(x=x, y=y, mode='lines', fill='tozeroy',
                         line=dict(color='mediumseagreen', width=3),
                         name='Triangular(4, 10, 6)'))

fig.update_layout(title="Triangular Distribution: Project Completion Time",
                  xaxis_title="Hours (x)", yaxis_title="Density f(x)",
                  template="plotly_white")
fig.show()

print(f"--- Numerical Results ---")
print(f"4. Expectation E[X]: {mean:.2f} hours")
print(f"5. Variance Var(X): {variance:.4f}")
print(f"8. P(X <= 5): {prob_8:.4f}")
print(f"9. P(5 <= X <= 8): {prob_9:.4f}")

--- Numerical Results ---
4. Expectation E[X]: 6.67 hours
5. Variance Var(X): 1.5556
8. P(X <= 5): 0.0833
9. P(5 <= X <= 8): 0.7500


10. Planning and Sensitivity Analysis:

Project Planning: The expectation ($6.67$ hours) provides a more realistic deadline than just using the "most likely" time ($6$ hours) because it accounts for the potential delays up to $10$ hours.

Sensitivity Analysis: By varying $b$ (the pessimistic time), managers can see how much the average completion time shifts. If the variance is high, it signals a high-risk task that might require more "buffer" time in the project schedule to prevent cascading delays.

#Q6) Expected Project Completion Time Using PERT

A project manager is estimating the time required to complete a critical software integration task. Let $X$ denote the number of days required to complete the task. Based on expert judgment, the shortest possible time is $a=8$ days, the longest possible time is $b=20$ days, and the most likely completion time is $c=11$ days. Assume that $X$ follows a PERT distribution with parameters $(a,b,c)$.

The PERT distribution is defined as a Beta distribution scaled to the interval $[a,b]$, with internal shape parameters

$$
\alpha = 1 + 4\left( \frac{c-a}{b-a} \right)
$$

and

$$
\beta = 1 + 4\left( \frac{b-c}{b-a} \right).
$$

Its probability density function is

$$
f(x)=\frac{(x-a)^{\alpha-1}(b-x)^{\beta-1}}{B(\alpha,\beta)(b-a)^{\alpha+\beta-1}}, \qquad a \leq x \leq b,
$$

where

$$
B(\alpha,\beta)=\int_0^1 t^{\alpha-1}(1-t)^{\beta-1},dt.
$$

Answer the following:

1. Compute the values of the shape parameters $\alpha$ and $\beta$.

2. Specify the support of $X$.

3. Write down the probability density function $f_X(x)$ for this model.

4. State the probability space $(\Omega,\mathfrak F,P)$ suitable for this continuous model.

5. Define $X$ as a random variable on this space.

6. Show that the expectation $E[X]$ is given by

$$
E[X]=\frac{a+4c+b}{6}.
$$

7. Compute the variance $\mathrm{Var}(X)$ using the approximation

$$
\mathrm{Var}(X)=\frac{(E[X]-a)(b-E[X])}{7}.
$$

8. Compute the approximate standard deviation $\sigma$.

9. Briefly explain why a PERT distribution may be more appropriate than a Triangular distribution in this setting byt comparing the two results.

10. Interpret the practical meaning of the values $a$, $b$, $c$, $E[X]$, and $\sigma$ in the context of project planning.


ANSWERS

---



1. Compute the shape parameters $\alpha$ and $\beta$:

Given $a=8, b=20, c=11$:$\alpha = 1 + 4\left(\frac{11 - 8}{20 - 8}\right) = 1 + 4\left(\frac{3}{12}\right) = 1 + 1 = \mathbf{2}$$\beta = 1 + 4\left(\frac{20 - 11}{20 - 8}\right) = 1 + 4\left(\frac{9}{12}\right) = 1 + 3 = \mathbf{4}$

2. Support of $X$:The support is the interval $[8, 20]$.

3. Probability Density Function $f_X(x)$:With $\alpha=2$ and $\beta=4$:$$f(x) = \frac{(x - 8)^{2-1}(20 - x)^{4-1}}{B(2,4)(20 - 8)^{2+4-1}} = \frac{(x - 8)(20 - x)^3}{B(2,4) \cdot 12^5}, \quad 8 \leq x \leq 20$$

4. Probability Space $(\Omega, \mathcal{F}, P)$:Sample Space ($\Omega$): The interval $[8, 20]$.$\sigma$-algebra ($\mathcal{F}$): The Borel $\sigma$-algebra $\mathcal{B}([8, 20])$.Probability Measure ($P$): Defined by $\int_A f_X(x) dx$ for any event $A \in \mathcal{F}$.

5. Definition of $X$:$X: \Omega \to \mathbb{R}$ is the identity random variable, $X(\omega) = \omega$, representing the number of days required to complete the task.

6. Expectation $E[X]$:$$E[X] = \frac{a + 4c + b}{6} = \frac{8 + 4(11) + 20}{6} = \frac{72}{6} = \mathbf{12} \text{ days}$$

7. Variance $\text{Var}(X)$:$$\text{Var}(X) = \frac{(12 - 8)(20 - 12)}{7} = \frac{4 \times 8}{7} = \frac{32}{7} \approx \mathbf{4.571}$$

8. Standard Deviation $\sigma$:$$\sigma = \sqrt{\text{Var}(X)} = \sqrt{4.571} \approx \mathbf{2.138} \text{ days}$$

In [7]:
import numpy as np
import plotly.graph_objects as go
from scipy.stats import triang, beta

# Parameters for Q6
a = 8
b = 20
c_mode = 11

# --- Triangular Distribution Calculation ---
c_scipy = (c_mode - a) / (b - a)
loc = a
scale = b - a

# --- PERT Distribution Calculation ---
alpha_pert = 1 + 4 * (c_mode - a) / (b - a)
beta_pert = 1 + 4 * (b - c_mode) / (b - a)

# Generate data points (extending range slightly for better visual)
x = np.linspace(0, 25, 500)
y_triang = triang.pdf(x, c=c_scipy, loc=loc, scale=scale)
y_pert = beta.pdf(x, alpha_pert, beta_pert, loc=a, scale=b-a)

# Create the Plotly figure
fig = go.Figure()

# Add Triangular Distribution (Dashed Royalblue)
fig.add_trace(go.Scatter(
    x=x,
    y=y_triang,
    mode='lines',
    name='Triangular Distribution',
    line=dict(color='royalblue', width=2, dash='dash')
))

# Add PERT Distribution (Solid Firebrick)
fig.add_trace(go.Scatter(
    x=x,
    y=y_pert,
    mode='lines',
    name='PERT Distribution',
    line=dict(color='firebrick', width=2)
))

# Layout updates
fig.update_layout(
    title=f'Comparison: Triangular vs PERT (a={a}, b={b}, c={c_mode})',
    xaxis_title='Time (days)',
    yaxis_title='Probability Density',
    template='plotly_white'
)

fig.show()

# Numerical results for the text cell
print(f"Expectation E[X]: {(a + 4*c_mode + b)/6:.2f}")
print(f"Variance Var(X): {(( (a + 4*c_mode + b)/6 ) - a) * (b - (a + 4*c_mode + b)/6 ) / 7:.4f}")

Expectation E[X]: 12.00
Variance Var(X): 4.5714


9. PERT vs. Triangular:A PERT distribution is often more appropriate than a Triangular distribution because it is "smoother." The Triangular model has a sharp peak at the mode ($c$), which is rarely physically realistic. PERT places more probability mass near the mean and reduces the impact of extreme outliers ($a$ and $b$), providing a more balanced "bell-like" curve that reflects natural human estimation error.

10. Context of Project Planning:

$a, b, c$: These are the "raw" expert inputs.

$E[X] = 12$: This is the date the manager should put on the schedule. Notice it is higher than the "most likely" time ($11$); it accounts for the "long tail" of potential delays up to 20 days.

$\sigma = 2.138$: This represents the uncertainty. A project manager can use this to say, "We are roughly 95% confident the task will be done within $12 \pm 2\sigma$ days" (approx. 8 to 16 days). This helps in calculating the "buffer" needed for the software integration.

#Q7) If the PERT Expectation is to Match Our Intuition

A project manager models the duration $X$ of a task by a PERT distribution with parameters $a$, $b$, and $c$, where $a$ is the minimum time, $b$ is the maximum time, and $c$ is the most likely time.

Suppose the minimum time is $a=5$ days and the most likely time is $c=11$ days. If the expected time computed from the PERT model is also to be the same as the most likely time:

1. Determine the value of the maximum time $b$.

2. What happens to the variance?

ANSWERS

---



1. Determine the value of the maximum time $b$:We are given that $a = 5$, $c = 11$, and the condition that the expected time $E[X]$ is equal to the most likely time $c$.The PERT expectation formula is:$$E[X] = \frac{a + 4c + b}{6}$$Setting $E[X] = c$, we get:$$c = \frac{a + 4c + b}{6}$$Multiplying by 6:$$6c = a + 4c + b$$Rearranging to solve for $b$:$$b = 2c - a$$Plugging in our values ($c = 11, a = 5$):$$b = 2(11) - 5 = 22 - 5 = \mathbf{17}$$So, the maximum time $b$ must be 17 days.

2. What happens to the variance?When $E[X] = c$, the distribution becomes perfectly symmetric.In the previous question (Q6), $c$ was closer to $a$ (left-leaning), which typically results in higher variance if the tail $b$ is very long.In this specific symmetric case, the "internal shape parameters" $\alpha$ and $\beta$ become equal.$$\alpha = 1 + 4\left(\frac{11-5}{17-5}\right) = 3, \quad \beta = 1 + 4\left(\frac{17-11}{17-5}\right) = 3$$Since $\alpha = \beta$, the distribution is a symmetric Beta distribution. The variance in this symmetric state is generally minimized for a fixed range $(b-a)$ compared to skewed versions with the same range, as the probability mass is concentrated evenly around the center rather than being stretched by a long pessimistic tail.

In [5]:
import numpy as np
import plotly.graph_objects as go
from scipy.stats import beta

# Parameters found
a, c = 5, 11
b = 17 # calculated result

# Calculate alpha and beta
alpha_val = 1 + 4 * ((c - a) / (b - a))
beta_val = 1 + 4 * ((b - c) / (b - a))

x = np.linspace(a, b, 500)
# Scaled Beta PDF (PERT)
y = beta.pdf((x - a) / (b - a), alpha_val, beta_val) / (b - a)

fig = go.Figure()

fig.add_trace(go.Scatter(
    x=x, y=y,
    mode='lines',
    fill='tozeroy',
    line=dict(color='darkcyan', width=3),
    name=f'Symmetric PERT (E[X]=c={c})'
))

# Marking the center
fig.add_vline(x=c, line_dash="dash", line_color="black",
              annotation_text="E[X] = c = 11")

fig.update_layout(
    title='Symmetric PERT Distribution where Expectation matches Mode',
    xaxis=dict(title='Days'),
    yaxis=dict(title='Density'),
    template='plotly_white'
)

fig.show()

print(f"Computed Parameters: alpha = {alpha_val}, beta = {beta_val}")
print(f"Range (b - a): {b - a} days")

Computed Parameters: alpha = 3.0, beta = 3.0
Range (b - a): 12 days
